usd/src/sdf_parser/Joint.cc

```diff
@@ -0,0 +1,402 @@
+
+        if (parentRotationTmp == ignition::math::Quaterniond::Identity)
+        {
+          parentRotationTmp = fixRotation * parentRotationTmp;
+        }
+        else
+        {
+          parentRotationTmp = ignition::math::Quaterniond(IGN_DTOR(-90),
+              IGN_PI, IGN_PI) * parentRotationTmp;
+        }
+
+        childRotationTmp = fixRotation * childRotationTmp;
+
+        parentRotation = pxr::GfQuatf(
+            parentRotationTmp.W(),
+            parentRotationTmp.X(),
+            parentRotationTmp.Y(),
+            parentRotationTmp.Z());
+        childRotation = pxr::GfQuatf(
+            childRotationTmp.W(),
+            childRotationTmp.X(),
+            childRotationTmp.Y(),
+            childRotationTmp.Z());
+        jointRevolute.CreateAxisAttr().Set(pxr::TfToken("X"));
+      }
+    }
+
+    _jointPrim.CreateLocalPos0Attr().Set(pxr::GfVec3f(
+          parentToJoint.Pos().X(),
+          parentToJoint.Pos().Y(),
+          parentToJoint.Pos().Z()));
+    _jointPrim.CreateLocalRot0Attr().Set(parentRotation);
+
+    _jointPrim.CreateLocalPos1Attr().Set(pxr::GfVec3f(
+          childToJoint.Pos().X(),
+          childToJoint.Pos().Y(),
+          childToJoint.Pos().Z()));
+    _jointPrim.CreateLocalRot1Attr().Set(childRotation);
+
+    return errors;
+  }
+
+  /// \brief Helper function to parse a SDF revolute joint to its USD
+  /// representation.
+  /// \param[in] _joint The SDF joint to parse
+  /// \param[in] _stage The stage that _joint should belong to. This must be a
+  /// valid, initialized stage.
+  /// \param[in] _path The path in _stage where the USD representation of _joint
+  /// will be defined.
+  /// \return UsdErrors, which is a vector of UsdError objects. Each UsdError
+  /// includes an error code and message. An empty vector indicates no error
+  /// occurred when converting _joint to its USD representation.
+  UsdErrors ParseSdfRevoluteJoint(const sdf::Joint &_joint,
+      pxr::UsdStageRefPtr &_stage, const std::string &_path)
+  {
+    UsdErrors errors;
+
+    auto usdJoint =
+      pxr::UsdPhysicsRevoluteJoint::Define(_stage, pxr::SdfPath(_path));
+
+    const auto axis = _joint.Axis();
+
+    if (axis->Xyz() == ignition::math::Vector3d::UnitX ||
+        axis->Xyz() == -ignition::math::Vector3d::UnitX)
+    {
+      usdJoint.CreateAxisAttr().Set(pxr::TfToken("X"));
+    }
+    else if (axis->Xyz() == ignition::math::Vector3d::UnitY ||
+        axis->Xyz() == -ignition::math::Vector3d::UnitY)
+    {
+      usdJoint.CreateAxisAttr().Set(pxr::TfToken("Y"));
+    }
+    else if (axis->Xyz() == ignition::math::Vector3d::UnitZ ||
+        axis->Xyz() == -ignition::math::Vector3d::UnitZ)
+    {
+      usdJoint.CreateAxisAttr().Set(pxr::TfToken("Z"));
+    }
+    else
+    {
+      errors.push_back(UsdError(sdf::Error(sdf::ErrorCode::ELEMENT_INVALID,
+            "Revolute joint [" + _joint.Name() + "] has an invalid axis.")));
```

Comment:
I wouldn't say that the axis is invalid, since SDFormat supports expressing an axis unit vector in any direction. I would say "Revolute joint [name] has specified an axis [x y z], but USD only supports integer values of [-1, 0, 1] when specifying joint axis unit vectors."